### PR TITLE
[Cypress] adding mochawesome report artifact to UI tests

### PR DESCRIPTION
**Pull request by mmartin24**
PR for: https://github.com/epinio/epinio-end-to-end-tests/issues/144

### Changes implemented
- Installed [mochawesome reporting tool](https://www.npmjs.com/package/mochawesome?activeTab=readme) in our Cypress test to improve quality reporting and easiness to locate errors by attaching screenshots and videos to exact errors.

### Explanation 
- CI will autogenerate an artifact named `mochawesome report` with all passing / failing tests like this one: [mochawesome-report.zip](https://github.com/epinio/epinio-end-to-end-tests/files/9703768/mochawesome-report.zip)
  - If succeeds, a green mark check with the content of the test will be displayed:    
  - If fails, the test will be marked in red, displaying the error, a screenshot an a video attached to the specific scenario:    
- The artifact needs to be downloaded an opened in a browser the `mochawesome.html` file  

Tested on CI  with a small sample of 1 test passing and 1 failng: https://github.com/epinio/epinio-end-to-end-tests/actions/runs/3178115569

### Small demo 

https://user-images.githubusercontent.com/37271841/193771119-e6a4c69f-7ceb-4d98-bbe3-effa28ff630f.mov

#### Note: 
The default version of mochawesome, creates a report without screenshots nor videos in it and calls the route where these ones are placed by default to 'bind' them to the main report to be later visualized. The problem is that it seems in our CI the screenshots and video files are not reachable aside from artifacts. This was workarounded by copying those files within the `mochawesome-report` and pointing the binding function to this route. Since we are not copying automatically screenshots and videos to the `mochawesome-report` locally, the reports will appear by default without them unless they are copied manually.

### CODE REVIEW

File: cypress/integration/unit_tests/menu.spec.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ describe('Menu testing', () => {
   });
 
   it('Verify Welcome Screen without Namespaces', () => {
-    cy.clickEpinioMenu('Namespaces');
+    cy.clickEpinioMenu('Namespacess');
```
Comment (thehejik) [inline cypress/integration/unit_tests/menu.spec.ts:31]: please revert this

File: .github/workflows/master_std_ui.yml [hunk 2]
```diff
@@ -150,6 +157,14 @@ jobs:
           name: cypress-videos
           path: cypress/videos
           retention-days: 7
+ 
+      - name: Upload Mochawesome report
+        uses: actions/upload-artifact@v2
+        if: always()
```
Comment (thehejik) [inline .github/workflows/master_std_ui.yml:163]: shouldn't be this also `if: failure()` ?

File: cypress/integration/unit_tests/menu.spec.ts (hunk 1 above)
Comment (mmartin24) [inline cypress/integration/unit_tests/menu.spec.ts:31]: typo. 
I forced this error to check how the implementation worked.
I will revert this.

File: .github/workflows/master_std_ui.yml (hunk 2 above)
Comment (mmartin24) [reply .github/workflows/master_std_ui.yml:163]: I thought it was needed always as we want also the good results to be displayed, but I may be wrong. 
I will try without it

File: cypress/integration/unit_tests/menu.spec.ts (hunk 1 above)
Comment (mmartin24) [reply cypress/integration/unit_tests/menu.spec.ts]: done

**Comment by mmartin24** [comment]
When tested in CI with all the runners in parallel, since there was a sole html file it was overriden all the time, leaving the report with the latest test solely: https://github.com/epinio/epinio-end-to-end-tests/actions/runs/3180302125

This was overcome by renaming an html per test https://github.com/epinio/epinio-end-to-end-tests/actions/runs/3181147403. 
image

**Comment by mmartin24** [comment]
Fixed another issue where, after an initial failed run and a successful second one, the step where screenshots were not attached would be marked as failed, raising a failed flag. This is fixed now and 2 runs, one failed and another one successful will be correctly marked with 2 rounds of files present in the artifact.
Example:
Round 1 failed: https://github.com/epinio/epinio-end-to-end-tests/actions/runs/3182560567/attempts/1
image

Round 2 successful:https://github.com/epinio/epinio-end-to-end-tests/actions/runs/3182560567
image

Both flags respond correctly to the change of situation.

**Review by thehejik — APPROVED** [review]
Good work, I haven't found any problem.

**Comment by mmartin24** [comment]
Thanks @thehejik 
Tested on several CI and working as expected. 
Merging

**Pull request merged**